src/usecases/associate-entity-use-case.ts

```diff
@@ -21,9 +22,15 @@ export const associateEntityUseCase = {
 		atlassianUserId,
 		connectInstallation,
 	}: AssociateEntityUseCaseParams): Promise<AtlassianDesign> => {
+		const figmaDesignId = FigmaDesignIdentity.fromFigmaDesignUrl(entity.url);
+
 		const [design, issue] = await Promise.all([
-			figmaService.fetchDesignByUrl(entity.url, atlassianUserId),
+			figmaService.fetchDesignById(figmaDesignId, atlassianUserId),
 			jiraService.getIssue(associateWith.id, connectInstallation),
+			associatedFigmaDesignRepository.upsert({
```

Comment:
What do you think about doing this upsert _after_ the design is sent to Data Depot? Otherwise if the ingestion fails, we will try to re-ingest this design if we receive a webhook for the containing file.

Reply:
That's a very good point, Justin! We should definitely do this.

Reply:
Fixed!